android-components/components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/OnDiskSitePermissionsStorage.kt

```diff
@@ -50,8 +50,14 @@ class OnDiskSitePermissionsStorage(
     /**
      * Persists the [sitePermissions] provided as a parameter.
      * @param sitePermissions the [sitePermissions] to be stored.
+     * @param private indicates if the [SitePermissions] belongs to a private session.
      */
-    override suspend fun save(sitePermissions: SitePermissions, request: PermissionRequest?) {
+    override suspend fun save(
+        sitePermissions: SitePermissions,
+        request: PermissionRequest?,
+        private: Boolean,
+    ) {
+        if (private) return
```

Comment:
Let's add a single line comment here and in the `update` method too. The line is a bit deceptive otherwise:

```suggestion
        if (private) return // never save private browsing site permissions
```